extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -282,11 +282,14 @@ private void createEmitter(BuildProducer<EmitterBuildItem> emitters, InjectionPo
 
     @BuildStep
     public void enableMetrics(BuildProducer<AnnotationsTransformerBuildItem> transformers,
-            Capabilities capabilities, ReactiveMessagingConfiguration configuration) {
-        boolean isMetricEnabled = capabilities.isPresent(Capability.METRICS) && configuration.metricsEnabled;
-        if (!isMetricEnabled) {
-            LOGGER.debug("Metric is disabled - vetoing the MetricDecorator");
-            // We veto the Metric Decorator
+            Optional<MetricsCapabilityBuildItem> metricsCapability,
+            ReactiveMessagingConfiguration configuration) {
+        boolean isMetricEnabled = metricsCapability.isPresent() && configuration.metricsEnabled;
+        boolean useMicrometer = isMetricEnabled && metricsCapability.get().metricsSupported(MetricsFactory.MICROMETER);
+        if (!isMetricEnabled || useMicrometer) {
```

Comment:
If it uses micrometer, how are the metrics reported?

Reply:
They aren't (yet). I was going to have another conversation with you about what metrics *should* be emitted w/ micrometer. (The metrics in this Decorator are only expected if MP Metrics is present, and I am not sure they're useful enough to preserve?)